### PR TITLE
Adopted refactored ingress protocol to prepare presets support

### PR DESCRIPTION
**Pull request by biglittlebigben**

### CODE REVIEW

File: pkg/media/params.go [hunk 1]
```diff
@@ -129,7 +137,17 @@ func isNilVideoParams(options *livekit.IngressVideoOptions) bool {
 		return true
 	}
 
-	if len(options.Layers) == 0 {
+	if _, ok := options.GetEncodingOptions().(*livekit.IngressVideoOptions_Preset); ok {
+		logger.Infow("presets unimplemented")
```
Comment (frostbyte73) [inline pkg/media/params.go:141]: nit: this might be a bit noisy

**Review by frostbyte73 — APPROVED** [review]

File: pkg/media/params.go (hunk 1 above)
Comment (biglittlebigben) [reply pkg/media/params.go:141]: This is meant to be temporary until I actually implement presets. Right now, presets are defined in the protocol, but you can't use them.

Comment (biglittlebigben) [reply pkg/media/params.go:141]: Would failing the ingress instead match the caller's expectations better?

Comment (frostbyte73) [reply pkg/media/params.go:141]: yeah, for now I'd return an `unimplemented` error

**Pull request merged**